packages/core/src/stylable-imports-tools.ts

```diff
@@ -0,0 +1,430 @@
+
+                if (!isCompRoot(importObj.defaultExport) && importObj.from.match(/\.css$/)) {
+                    diagnostics.warn(decl, parseImportMessages.DEFAULT_IMPORT_IS_LOWER_CASE(), {
+                        word: importObj.defaultExport,
+                    });
+                }
+                break;
+            case valueMapping.named:
+                {
+                    const { keyframesMap, namedMap } = parseNamed(decl.value, decl, diagnostics);
+                    importObj.named = namedMap;
+                    importObj.keyframes = keyframesMap;
+                }
+                break;
+            default:
+                diagnostics.warn(decl, parseImportMessages.ILLEGAL_PROP_IN_IMPORT(decl.prop), {
+                    word: decl.prop,
+                });
+                break;
+        }
+    });
+
+    if (!importObj.from) {
+        diagnostics.error(rule, parseImportMessages.FROM_PROP_MISSING_IN_IMPORT());
+    }
+    return importObj;
+}
+
+function createPseudoImportProps(
+    item: Partial<Pick<Imported, 'named' | 'keyframes' | 'defaultExport' | 'request'>>
+) {
+    const nodes = [];
+    const named = generateNamedValue(item);
+    if (item.request) {
+        nodes.push(decl({ prop: '-st-from', value: JSON.stringify(item.request) }));
+    }
+    if (item.defaultExport) {
+        nodes.push(
+            decl({
+                prop: '-st-default',
+                value: item.defaultExport,
+            })
+        );
+    }
+    if (named.length) {
+        nodes.push(
+            decl({
+                prop: '-st-named',
+                value: named.join(', '),
+            })
+        );
+    }
+
+    return {
+        selector: ':import',
+        nodes,
+    };
+}
+
+function patchDecls(node: Rule, named: string[], pseudoImport: Imported) {
+    const namedDecls: Declaration[] = [];
+    const defaultDecls: Declaration[] = [];
+    node.walkDecls((decl) => {
+        if (decl.prop === '-st-named') {
+            decl.assign({ value: named.join(', ') });
+            namedDecls.push(decl);
+        } else if (decl.prop === '-st-default') {
+            decl.assign({ value: pseudoImport.defaultExport });
+            defaultDecls.push(decl);
+        }
+    });
+    return { defaultDecls, namedDecls };
+}
+
+function ensureSingleDecl(decls: Declaration[], node: Rule, prop: string, value: string) {
+    if (!decls.length) {
+        node.append(decl({ prop, value }));
+    } else if (decls.length > 1) {
+        // remove duplicates keep last one
+        for (let i = 0; i < decls.length - 1; i++) {
+            decls[i].remove();
+        }
+    }
+}
+
+function getNamedImportParts(named: [string, string][]) {
+    const parts: string[] = [];
+    for (const [as, name] of named) {
+        if (as === name) {
+            parts.push(name);
+        } else {
+            parts.push(`${name} as ${as}`);
+        }
+    }
+
+    return parts;
+}
+
+type ImportPatch = {
```

Comment:
I understand that the fields are copies of the already existing `Imported` interface, but `request` would have been more clear as `from`, `defaultExport` should have been `defaultImport`